### PR TITLE
fix: bimatchranker passing param

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
Caused by https://github.com/jina-ai/jina/pull/1947
Several ways to solve this issue. Not sure which is the best. This issue caused the error of Tumblr-search.

1. See this PR
2. ` query_required_keys = ('length', )
       match_required_keys = ('length', )

    def __init__(self, d_miss: Optional[Union[int, float]] = None, *args, **kwargs):
        super().__init__(self.query_required_keys, self. match_required_keys, *args, **kwargs)`

### CODE REVIEW

File: rankers/BiMatchRanker/__init__.py [hunk 1]
```diff
@@ -58,11 +57,13 @@ class BiMatchRanker(Chunk2DocRanker):
 
     .. warning:: Here we suppose that the smaller chunk score means the more similar.
     """
-    query_required_keys = ('length', )
-    match_required_keys = ('length', )
 
-    def __init__(self, d_miss: Optional[Union[int, float]] = None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    # query_required_keys = ('length', )
```
Comment (JoanFM) [inline rankers/BiMatchRanker/__init__.py:61]: remove these ones and it is good for me

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: rankers/BiMatchRanker/__init__.py (hunk 1 above)
Comment (Yongxuanzhang) [reply rankers/BiMatchRanker/__init__.py]: ✅

File: rankers/BiMatchRanker/__init__.py [hunk 2]
```diff
@@ -58,11 +57,10 @@ class BiMatchRanker(Chunk2DocRanker):
 
     .. warning:: Here we suppose that the smaller chunk score means the more similar.
     """
-    query_required_keys = ('length', )
-    match_required_keys = ('length', )
 
-    def __init__(self, d_miss: Optional[Union[int, float]] = None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, query_required_keys: Optional = ('length',), match_required_keys: Optional = ('length',),
+                 d_miss: Optional[Union[int, float]] = None, *args, **kwargs):
```
Comment (bwanglzu) [inline rankers/BiMatchRanker/__init__.py:62]: docstrings needs to be updated as well

**Review by bwanglzu — CHANGES_REQUESTED** [review]

File: rankers/BiMatchRanker/__init__.py (hunk 2 above)
Comment (Yongxuanzhang) [reply rankers/BiMatchRanker/__init__.py:64]: 👍

File: rankers/BiMatchRanker/__init__.py [hunk 3]
```diff
@@ -52,17 +51,18 @@ class BiMatchRanker(Chunk2DocRanker):
...
     :param kwargs: Additional keyword arguments
 
     .. warning:: Here we suppose that the smaller chunk score means the more similar.
     """
-    query_required_keys = ('length', )
-    match_required_keys = ('length', )
 
-    def __init__(self, d_miss: Optional[Union[int, float]] = None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, query_required_keys: Optional = ('length',), match_required_keys: Optional = ('length',),
```
Comment (JoanFM) [inline rankers/BiMatchRanker/__init__.py:63]: I think these should not be parameters, but they should be passed directly to the `super`constructor.

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: rankers/BiMatchRanker/__init__.py (hunk 3 above)
Comment (Yongxuanzhang) [reply rankers/BiMatchRanker/__init__.py:63]: In that case, the following one may be better? It depends on if we want to expose these params to users and the intention you refactored before,
`query_required_keys = ('length', )
match_required_keys = ('length', )

def init(self, d_miss: Optional[Union[int, float]] = None, *args, **kwargs):
super().init(self.query_required_keys, self. match_required_keys, *args, **kwargs)`

Comment (JoanFM) [reply rankers/BiMatchRanker/__init__.py:63]: For now in this case, I think we do not want to expose it to users.

Comment (Yongxuanzhang) [reply rankers/BiMatchRanker/__init__.py:63]: Sure, I will change to the second one. Or any other suggestions? 😄

Comment (JoanFM) [reply rankers/BiMatchRanker/__init__.py:63]: The second one, and passed by value if possible I would say

Comment (Yongxuanzhang) [reply rankers/BiMatchRanker/__init__.py]: Sure. Direct pass values now

File: rankers/BiMatchRanker/__init__.py [hunk 4]
```diff
@@ -58,11 +57,9 @@ class BiMatchRanker(Chunk2DocRanker):
 
     .. warning:: Here we suppose that the smaller chunk score means the more similar.
     """
-    query_required_keys = ('length', )
-    match_required_keys = ('length', )
 
     def __init__(self, d_miss: Optional[Union[int, float]] = None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+        super().__init__(('length', ), ('length', ), *args, **kwargs)
```
Comment (JoanFM) [inline rankers/BiMatchRanker/__init__.py:62]: can u pass named arguments? like this if arguments change order in the future, this will still work

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: rankers/BiMatchRanker/__init__.py (hunk 4 above)
Comment (Yongxuanzhang) [reply rankers/BiMatchRanker/__init__.py:62]: Like this?
`super().__init__(query_required_keys=('length', ), match_required_keys=('length', ), *args, **kwargs)`

**Review by JoanFM — APPROVED** [review]

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Pull request merged**